frontend/tests/e2e/login.spec.ts

```diff
@@ -0,0 +1,114 @@
+import { expect, test } from "@playwright/test";
+
+const BASE_URL = "http://127.0.0.1:3000";
+
+test.describe("Login Page Redirect", () => {
+  test.beforeEach(async ({ page }) => {
+    // Clear session storage before each test
+    await page.goto(`${BASE_URL}/`);
+    await page.evaluate(() => {
+      if (window.sessionStorage) {
+        window.sessionStorage.clear();
+      }
+    });
+    await page.context().addCookies([
+      {
+        name: "_ff",
+        value: JSON.stringify({ authOn: true }),
+        domain: "127.0.0.1",
+        path: "/",
+      },
+    ]);
+
+    // Prevent navigation to external sites by redirecting back to base URL
+    await page.route("**/*", async (route) => {
+      const url = route.request().url();
+      if (
+        !url.startsWith(BASE_URL) &&
+        !url.startsWith("http://127.0.0.1:8080")
+      ) {
+        await route.fulfill({
+          status: 200,
+          contentType: "text/html",
+          body: `<html><head><meta http-equiv="refresh" content="0;url=${BASE_URL}/"></head></html>`,
+        });
+      } else {
+        await route.continue();
+      }
+    });
+  });
+
+  test("should redirect to home page when no redirect URL is stored", async ({
+    page,
+  }) => {
+    await page.goto(`${BASE_URL}/login`);
+    await expect(page).toHaveURL(`${BASE_URL}/`);
+  });
+
+  test("should redirect to stored URL after login", async ({ page }) => {
+    await page.evaluate(() => {
+      sessionStorage.setItem("login-redirect", "/opportunities");
+    });
+    await page.goto(`${BASE_URL}/login`);
+    await expect(page).toHaveURL(`${BASE_URL}/opportunities`);
+  });
+
+  test("should redirect to home page when stored URL is empty", async ({
+    page,
+  }) => {
+    await page.evaluate(() => {
+      sessionStorage.setItem("login-redirect", "");
+    });
+    await page.goto(`${BASE_URL}/login`);
+    await expect(page).toHaveURL(`${BASE_URL}/`);
+  });
+
+  test("should redirect to home page when stored URL is external", async ({
+    page,
+  }) => {
+    await page.evaluate(() => {
+      sessionStorage.setItem("login-redirect", "https://external.com");
+    });
+    await page.goto(`${BASE_URL}/login`);
+    await expect(page).toHaveURL(`${BASE_URL}/`);
+  });
+
+  test('should display "Redirecting..." text while redirecting', async ({
+    page,
+  }) => {
+    await page.evaluate(() => {
+      sessionStorage.setItem("login-redirect", "/opportunities");
+    });
+    await page.goto(`${BASE_URL}/login`);
+    await expect(page.getByText("Redirecting...")).toBeVisible();
+  });
+
+  test("should handle redirect after successful login flow", async ({
+    page,
+  }) => {
+    // Navigate to opportunities page
+    await page.goto(`${BASE_URL}/opportunities`);
```

Comment:
it doesn't look like this test is finished, which is ok. So far we have not been able to get the e2e suite to handle the post login redirect successfully, and have a separate ticket to deal with that https://github.com/HHS/simpler-grants-gov/issues/3459

for now you can delete this test